src/cc/frontends/clang/b_frontend_action.cc

```diff
@@ -811,10 +811,20 @@ bool BTypeVisitor::VisitFunctionDecl(FunctionDecl *D) {
   if (fe_.is_rewritable_ext_func(D)) {
     current_fn_ = string(D->getName());
     string bd = rewriter_.getRewrittenText(expansionRange(D->getSourceRange()));
-    fe_.func_src_.set_src(current_fn_, bd);
+    auto func_info = fe_.prog_func_info_.add_func(current_fn_);
+    if (!func_info) {
+      error(GET_BEGINLOC(D), "redefinition of existing function");
+      return false;
+    }
+    func_info->src_ = bd;
     fe_.func_range_[current_fn_] = expansionRange(D->getSourceRange());
-    string attr = string("__attribute__((section(\"") + BPF_FN_PREFIX + D->getName().str() + "\")))\n";
-    rewriter_.InsertText(real_start_loc, attr);
+    if (SectionAttr *A = D->getAttr<SectionAttr>()) {
+      func_info->section_ = A->getName().str();
```

Comment:
`section_` can also be set when object is loaded (and is - see `notifyObjectLoaded`) 

I'm inclined to remove this in favor of `notifyObjectLoaded` as that's at the end of the compilation process and therefore leaves less (no?) room for unexpected section changes that the FrontendAction might not know about.